### PR TITLE
Image repository commands

### PR DESCRIPTION
**Pull request by jakolehm**

### CODE REVIEW

File: lib/kontena/plugin/cloud/image/show_command.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  parameter "NAME", "Image repository name"
+
+  def execute
+    repo = Kontena::Cli::Models::ImageRepo.new(image_registry_client.get("/repositories/#{name}")['data'])
+    puts "#{name}:"
+    puts "  created: #{ time_ago(repo.created_at.to_i)}"
+    puts "  updated: #{ time_ago(repo.updated_at.to_i)}"
+    puts "  pulls: #{repo.pulls}"
+    puts "  pushs: #{repo.pushs}"
```
Comment (kke) [inline lib/kontena/plugin/cloud/image/show_command.rb]: `pushes`?

Comment (jakolehm) [reply lib/kontena/plugin/cloud/image/show_command.rb]: yes 👍

**Comment by kke** [comment]
The repository list endpoints support search using `?query=xxx` but it's probably not necessary in the cli at this point.

File: lib/kontena/plugin/cloud/image/login_command.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      token = cloud_client.post("/user/personal_access_tokens", { data: data })['data']
+    end
+
+    success = nil
+    spinner "Logging in to the Kontena Cloud Image Registry" do
+      success = system("echo '#{token.dig('attributes', 'access-token')}' | docker login -u #{current_account.username} --password-stdin #{image_distribution_url} > /dev/null")
+    end
+    if success
+      puts ""
+      puts "  Login succeeded, you should be now able to push and pull images using docker cli from #{image_distribution_url}"
```
Comment (kke) [inline lib/kontena/plugin/cloud/image/login_command.rb]: `should be now able` -> `should now be able`

Comment (kke) [reply lib/kontena/plugin/cloud/image/login_command.rb]: Maybe add a hint how to address the registry in docker, `docker tag` and `docker push xyz/xxx` and how to specify it in the stack.yml?

Comment (kke) [reply lib/kontena/plugin/cloud/image/login_command.rb]: (and how to configure the platform/master/agent to use it)

**Comment by kke** [comment]
image

Complains about `--password-stdin` but says `[done]`

**Comment by kke** [comment]
image

File: lib/kontena/plugin/cloud/image/login_command.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    success = spinner "Logging in to the Kontena Cloud Image Registry" do |spin|
+      pass = token.dig('attributes', 'access-token')
+
+      if `docker login --help`['--password-stdin']
+        output, stderr, status = Open3.capture3("docker login -u %s --password-stdin %s" % [current_account.username, image_distribution_url].map(&:shellescape), :stdin_data => pass)
+        unless status.success?
+          exit_with_error stderr
+        end
+      else
+        pass = 'foo'
```
Comment (kke) [inline lib/kontena/plugin/cloud/image/login_command.rb]: Does not work if someone's password is not "foo"

Comment (jnummelin) [reply lib/kontena/plugin/cloud/image/login_command.rb]: ah, some left-over from failure injection testing 🤦‍♂️

**Pull request merged**